src/RoboSaverVirtualModule.sol

```diff
@@ -157,7 +174,12 @@ contract RoboSaverVirtualModule {
     /// @return adjustPoolNeeded True if there is a deficit or surplus; false otherwise
     /// @return execPayload The payload of the needed transaction
     function checker() external view returns (bool adjustPoolNeeded, bytes memory execPayload) {
-        if (_isTxQueued()) return (false, bytes("Transaction in queue, wait for it to be executed"));
+        if (_isExternalTxQueued()) return (false, bytes("Transaction in queue, wait for it to be executed"));
+
+        /// @notice checks if there is a transaction queued up in the delay module by the virtual module itself
+        if (txQueueData.nonce != 0) {
```

Comment:
it looks like `queueNonce` in `Delay` starts at 0. how to handle if we are the first tx?

Reply:
if we are the first one pushing a tx the nonce in the delay module will be at 1 for `queueNonce` and in storage also will be `txQueueData.nonce  = 1`

Reply:
are you sure? if `queueNonce` is `0` when `execTransactionFromModule` is called, it looks like it calcing the hash first, before incrementing it:
```
    function execTransactionFromModule(
        address to,
        uint256 value,
        bytes calldata data,
        Enum.Operation operation
    ) public override moduleOnly returns (bool success) {
        bytes32 hash = getTransactionHash(to, value, data, operation);
        txHash[queueNonce] = hash;
        txCreatedAt[queueNonce] = block.timestamp;
        emit TransactionAdded(queueNonce, hash, to, value, data, operation);
        queueNonce++;
        success = true;
    }
```

Reply:
in our case we are reading the delay module storage after `execTransactionFromModule` is called. [see](https://github.com/gnosisguild/zodiac-modifier-delay/blob/main/contracts/Delay.sol#L138). it will have being then incremented

Reply:
ah yes we read it after calling `execTransactionFromModule`